### DISCUSSION TITLE
Custom extensions

### DISCUSSION THREAD

**Discussion by dadyarri** [cookiecutter/cookiecutter#1557] [Q&A] [answered]
How to to use custom extensions in template directory without changing pythonpath?

In my template (https://github.com/dadyarri/lab-work-template) i have extensions folder in there one python file with jinja extension, which used in one of the generated files.

cookiecutter.json have correct link to this extension:

https://github.com/dadyarri/lab-work-template/blob/fc4a755465c68b2791851be659e5e50d29021825/cookiecutter.json#L13

but when i try to run cookiecutter with this substitution `{% cookiecutter.file_name|to_camel %}` i got this:

```python
  File ".\tests\test_{{cookiecutter.file_name}}.py", line 3, in template
jinja2.exceptions.TemplateSyntaxError: Encountered unknown tag 'cookiecutter'.
  File ".\tests\test_{{cookiecutter.file_name}}.py", line 3
    class Test{% cookiecutter.file_name|to_camel %}:
```

 this with this substitution `{{ cookiecutter.file_name|to_camel }}`

```python
File ".\tests\test_{{cookiecutter.file_name}}.py", line 3, in template
jinja2.exceptions.TemplateAssertionError: No filter named 'to_camel'.
  File ".\tests\test_{{cookiecutter.file_name}}.py", line 3
    class Test{{ cookiecutter.file_name|to_camel }}:
```

**Comment by simobasso** [comment, marked as answer]
Hey @dadyarri 👋 ,

Cookiecutter needs to import the extension module; if the module is not in your python path, it will raise an `ImportError` under the hood.

So the options, IMHO, are:

- change the python path
- install your template as a package and use the full package like es. : `"_extensions": ["lab_work_template.extensions.to_camel.ToCamelCaseExtension"]`
- wait for https://github.com/cookiecutter/cookiecutter/pull/1240/files

We hope to include this feature shortly.

Thanks 🙏
